File: app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.java
package deakin.gopher.guardian.view.general;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import deakin.gopher.guardian.R;

public class Homepage4nurse extends AppCompatActivity {

  Button tasksButton, settingsButton, signOutButton;

  @SuppressLint("MissingInflatedId")
  @Override
  protected void onCreate(final Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_homepage4nurse);

    tasksButton = findViewById(R.id.tasksButton_nurse);
    settingsButton = findViewById(R.id.settingsButton_nurse);
    signOutButton = findViewById(R.id.sighOutButton_nurse);

    // patient list button
      tasksButton.setOnClickListener(
        view -> {
          final Intent medicalDiagnosticsActivityIntent =
              new Intent(Homepage4nurse.this, PatientListActivity.class);
          startActivity(medicalDiagnosticsActivityIntent);
        });

    // settings button
    settingsButton.setOnClickListener(
        view -> {
          final Intent medicalDiagnosticsActivityIntent =
              new Intent(Homepage4nurse.this, Setting.class);
          startActivity(medicalDiagnosticsActivityIntent);
        });

    // sign out button
    signOutButton.setOnClickListener(
        view -> {
          final Intent medicalDiagnosticsActivityIntent =
              new Intent(Homepage4nurse.this, LoginActivity.class);
          startActivity(medicalDiagnosticsActivityIntent);
        });
  }
}

File: app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.kt
package deakin.gopher.guardian.view.general

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import deakin.gopher.guardian.R
import deakin.gopher.guardian.model.login.SessionManager
import deakin.gopher.guardian.services.EmailPasswordAuthService
import deakin.gopher.guardian.services.NavigationService

class Homepage4nurse : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_homepage4nurse)

        val user = SessionManager.getCurrentUser()

        val titleText: TextView =
            findViewById(R.id.medicalDiagnosticsTitleTextView)

        titleText.append(" ${user.name.split(" ").getOrNull(0) ?: ""}")

        // Existing buttons
        val patientsButton: Button =
            findViewById(R.id.patientsButton_nurse)

        val settingsButton: Button =
            findViewById(R.id.settingsButton_nurse)

        val signOutButton: Button =
            findViewById(R.id.sighOutButton_nurse)

        // New Patient Activities button
        val patientActivitiesButton: Button =
            findViewById(R.id.patientActivitiesButton_nurse)

        // Patients button
        patientsButton.setOnClickListener {
            NavigationService(this).onLaunchPatientList()
        }

        // Settings button
        settingsButton.setOnClickListener {
            NavigationService(this).onSettings()
        }

        // Patient Activities button
        patientActivitiesButton.setOnClickListener {

            // Temporary navigation
            // Replace PatientActivitiesActivity later if needed

            startActivity(
                Intent(
                    this,
                    PatientActivitiesActivity::class.java
                )
            )
        }

        // Sign out button
        signOutButton.setOnClickListener {
            EmailPasswordAuthService.signOut(this)
            finish()
        }
    }
}